src/Makefile.am

```diff
@@ -26,6 +26,9 @@ include @top_srcdir@/src/Makefile.theories
 lib_LTLIBRARIES = libcvc4.la
 
 libcvc4_la_LDFLAGS = -version-info $(LIBCVC4_VERSION)
+@mk_if@eq ($(OS),Windows_NT)
+libcvc4_la_LDFLAGS += -no-undefined
```

Comment:
It might be a good idea to use this flag on all platforms. You mentioned in your email that there might be potential problems with macOS, I'll try that.